Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemGroupsBusinessService.cs

```diff
@@ -82,17 +82,19 @@ await sourceContestProblemGroups
             return ServiceResult.Success;
         }
 
-        public async Task ReevaluateProblemsAndProblemGroupsByOrderBy(int contestId)
+        public async Task ReevaluateProblemsAndProblemGroupsByOrderBy(int contestId, ProblemGroup problemGroup)
         {
             var problemGroups = this.problemGroupsData.GetAllVisibleByContest(contestId);
 
             await this.problemGroupsOrderableService.ReevaluateOrderBy(problemGroups);
 
+            // We detach the existing entity, in order to avoid tracking exception on Update.
+            this.problemGroupsData.Detach(problemGroup);
+
             var problems = problemGroups.SelectMany(p => p.Problems)
                                                             .Where(p => !p.IsDeleted);
 
             await this.problemsOrderableService.ReevaluateOrderBy(problems);
```

Comment:
Why the `Detach` is after the problemGroupsOrderableService.ReevaluateOrderBy(problemGroups)? Do we need it all? Do we need it just for the Problems, and not for the ProblemGroups?

Reply:
The problem is that the current problemGroup tracks the problem. It is after problemGroupsOrderableService.ReevaluateOrderBy(problemGroups), because otherwise, the entity is not being tracked and this causes invalid reevaluation of the problemGroups.